### PR TITLE
Added ignore-not-found flag for kubectl patch

### LINKED ISSUE #108798: kubectl patch doesn't have --ignore-not-found
/sig cli

### What happened?

I use `--ignore-not-found=true` for `delete/stop/get/etc` but it looks like that flag isn't available for `kubectl patch`

### What did you expect to happen?

I want to be able to use `--ignore-not-found=true` in `kubectl patch`

### How can we reproduce it (as minimally and precisely as possible)?

```
$ kubectl patch crd/xxx.com -p '{"metadata":{"finalizers":[]}}' --type=merge --kubeconfig kind.yaml
Error from server (NotFound): customresourcedefinitions.apiextensions.k8s.io "xxx.com" not found

$ kubectl patch crd/xxx.com -p '{"metadata":{"finalizers":[]}}' --type=merge --kubeconfig kind.yaml --ignore-not-found=true
error: unknown flag: --ignore-not-found
See 'kubectl patch --help' for usage.
```

### Anything else we need to know?

_No response_

### Kubernetes version

$ kubectl version --kubeconfig kind.yaml 
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5", GitCommit:"c285e781331a3785a7f436042c65c5641ce8a9e9", GitTreeState:"clean", BuildDate:"2022-03-16T15:58:47Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.10", GitCommit:"a7a32748b5c60445c4c7ee904caf01b91f2dbb71", GitTreeState:"clean", BuildDate:"2022-03-07T09:03:24Z", GoVersion:"go1.16.14", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.23) and server (1.21) exceeds the supported minor version skew of +/-1

### Cloud provider

N/A

### OS version

<details>
```console
# On Linux:
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.6 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic</details>


### Install tools

_No response_

### Container runtime (CRI) and version (if applicable)

_No response_

### Related plugins (CNI, CSI, ...) and versions (if applicable)

_No response_

### PR DESCRIPTION
**Pull request by sreeram-venkitesh**
#### What type of PR is this?

/kind feature 

#### What this PR does / why we need it:
This PR adds the optional `--ignore-not-found` flag to `kubectl patch` command. This makes the `patch` command consistent with other commands like `get`, and `delete`

#### Which issue(s) this PR fixes:

Fixes #108798 

#### Does this PR introduce a user-facing change?

```release-note
The `kubectl patch` command now supports the optional `--ignore-not-found` flag
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by sreeram-venkitesh** [comment]
CC: @mpuckett159

File: staging/src/k8s.io/kubectl/pkg/cmd/patch/patch_test.go [hunk 1]
```diff
@@ -240,3 +241,49 @@ func TestPatchSubresource(t *testing.T) {
...
+	cmd := NewCmdPatch(tf, stream)
+	cmd.Flags().Set("namespace", "test")
+	cmd.Flags().Set("ignore-not-found", "true")
+	cmd.Flags().Set("patch", `{"spec":{"type":"NodePort"}}`)
+	cmd.Flags().Set("output", "name")
+	cmd.Run(cmd, []string{"services/nonexistentservice"})
+
+	if buf.String() != "" {
+		t.Errorf("unexpected output: %s", buf.String())
+	}
```
Comment (mpuckett159) [inline staging/src/k8s.io/kubectl/pkg/cmd/patch/patch_test.go:288]: Can we add a test for the flag used to patch something that does exist in addition to testing against something that doesn't exist?

**Comment by sreeram-venkitesh** [comment]
Closed as not planned. The PR was discussed in the SIG CLI meeting and the SIG decided not to go ahead with this change since adding consistency with other commands like `kubectl get` and `kubectl delete` doesn't add enough value to justify the change. `kubectl patch` is meant to update only a single resource at a time, and the `--ignore-not-found` flag is generally used in scenarios where you have to update multiple resources together. In such cases, as with any case of automation (which was my requirement), it is recommended to use `kubectl apply` instead of `kubectl patch` since `apply` would make the change declaratively and would keep a record of the previous history of the resource.

Watch the recording of the SIG CLI meeting where this was discussed [here](https://youtu.be/BSMhMCJ7rIo).

**Pull request closed without merging**